modules/IntegrationTesting/src/scenarioexecution.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+#include <iomanip>
+#include "scenarioexecution.hpp"
+#include "rclcpp/wait_for_message.hpp"
+
+using namespace std::chrono_literals;
+
+/**
+ * @brief An integration test for scenario execution. This test will run a scenario execution for one object,
+ * and check that all states are correct and that it follows the trajectory.
+ * 
+ */
+ScenarioExecution::ScenarioExecution() : IntegrationTesting(testName) {
+	RCLCPP_INFO(get_logger(), "Started integration test: %s", testName.c_str());
+	getObjectTrajectoryClient = this->create_client<atos_interfaces::srv::GetObjectTrajectory>("/atos/get_object_trajectory");
+	runIntegrationTest();
+}
+
+
+/**
+ * @brief Destructor.
+ * 
+ */
+ScenarioExecution::~ScenarioExecution() {}
+
+
+/**
+ * @brief Run the integration test. This integration test will initialize a scenario, connect to object 1,
+ * arm the object, and then start it. It will check that all states are correct, and that the object follows
+ * the trajectory.
+ * 
+ */
+void ScenarioExecution::runIntegrationTest() {
+	RCLCPP_INFO(get_logger(), "Running scenario");
+	auto msg = std_msgs::msg::Empty();
+
+	initPub->publish(msg);
+	std::this_thread::sleep_for(std::chrono::seconds(1)); // sleep to allow the object to initialize
+	checkState(initTopic);
+	
+	connectPub->publish(msg);
+	std::this_thread::sleep_for(std::chrono::seconds(1)); // sleep to allow the object to connect
+	checkState(connectTopic);
+
+	armPub->publish(msg);
+	std::this_thread::sleep_for(std::chrono::seconds(1)); // sleep to allow the object to arm
+	checkState(armTopic);
+
+	startPub->publish(msg);
+	std::this_thread::sleep_for(std::chrono::seconds(1)); // sleep to allow the object to start
+	checkState(startTopic);
+
+	checkTrajectory();
+	printResult();
+}	
+
+
+/**
+ * @brief Get the x- and y-coordinates of the trajectory points for object 1.
+ * 
+ * @return std::vector<std::pair<double, double>> Trajectory points.
+ */
+std::vector<std::pair<double, double>> ScenarioExecution::getTrajectoryPoints() {
+	auto request = std::make_shared<atos_interfaces::srv::GetObjectTrajectory::Request>();
+	request->id = 1;
+	std::shared_ptr<atos_interfaces::srv::GetObjectTrajectory::Response> response;
+	this->callService(1000ms, getObjectTrajectoryClient, request, response);
+
+	std::vector<std::pair<double, double>> trajectory;
+	for (const auto& t : response->trajectory.points) {
+		trajectory.push_back(std::make_pair(t.pose.position.x, t.pose.position.y));
+	}
+	return trajectory;
+}
+
+
+/**
+ * @brief Check that the object ends up near the end point of the trajectory.
+ * 
+ */
+void ScenarioExecution::checkTrajectory() {
```

Comment:
Could this check be also along the trajectories? Maybe not for this PR but as a future improvement

Reply:
Yes, I think it's a good idea to extend this in the future. But I think it's good enough for now.